### PR TITLE
[13.0][FIX] sale_coupon_product_management constrains

### PR DESCRIPTION
**Pull request by jcoux**
Fix constrains to have no duplicate products on coupon programs.

### CODE REVIEW

File: sale_coupon_product_management/models/sale_coupon_program.py [hunk 1]
```diff
@@ -69,19 +69,24 @@ def _check_program_options(self):
 
     def _check_no_product_duplicate(self):
         for rec in self:
```
Comment (SilvioC2C) [inline sale_coupon_product_management/models/sale_coupon_program.py]:
```suggestion
        for rec in self.filtered("discount_line_product_id"):
```
This way you don't have to change the rest of the constrain method :smile:

**Review by SilvioC2C — APPROVED** [review]
LGTM, just a minor suggestion

File: sale_coupon_product_management/models/sale_coupon_program.py (hunk 1 above)
Comment (jcoux) [reply sale_coupon_product_management/models/sale_coupon_program.py]: Thanks, done!

**Review by damdam-s — APPROVED** [review]

**Comment by rousseldenis** [comment]
@jcoux Maybe a test for this change ?

**Pull request merged**